### PR TITLE
DWB implementation using real costmap

### PR DESCRIPTION
**Pull request by crdelsey**
This change finishes out the integration of a DWB controller into the task model and links to the real costmap

It might be helpful to review by going through the commit history. The first change is the DWB task integration. The rest are changes to get DWB to compile and link cleanly with costmap.

### CODE REVIEW

File: src/control/nav2_controller_dwb/src/dwb_controller.cpp [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  nav2_tasks::FollowPathResult result;
+  setResult(result);
+
+  return TaskStatus::SUCCEEDED;
+}
+
+void DwbController::publishVelocity(const nav_2d_msgs::msg::Twist2DStamped & velocity)
+{
+  auto cmd_vel = nav_2d_utils::twist2Dto3D(velocity.velocity);
+  vel_pub_->publish(cmd_vel);
```
Comment (mjeronimo) [inline src/control/nav2_controller_dwb/src/dwb_controller.cpp]: The publish of the velocities could be via a robot method, where the robot class has the publisher (future TODO).

File: src/control/nav2_controller_dwb/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  ${Boost_INCLUDE_DIRS}
+)
+
+set(library_name ${executable_name}_core)
+
+add_library(${library_name}
+  src/dwb_controller.cpp
+)
+
+set(executable_name nav2_controller_dwb)
```
[the remark this answers is not in the document]
Comment (crdelsey) [reply src/control/nav2_controller_dwb/CMakeLists.txt]: I can use `executable_name` before defining it, but it evaluates to `""` so my library is misnamed. Good catch. I'll fix this.

File: src/control/nav2_controller_dwb/src/dwb_controller.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    tf2_ros::TransformListener tfListener(tfBuffer);
+    cm_ = std::make_shared<costmap_2d::Costmap2DROS>("local_costmap", tfBuffer);
+    auto nh = shared_from_this();
+    odom_sub_ = std::make_shared<nav_2d_utils::OdomSubscriber>(*this);
+    vel_pub_ = this->create_publisher<geometry_msgs::msg::Twist>("cmd_vel", 1);
+    planner_.initialize(nh, shared_ptr<tf2_ros::Buffer>(&tfBuffer), cm_);
+    planner_.setPlan(path);
+    while (true) {
+      nav_2d_msgs::msg::Pose2DStamped pose2d;
+      if (!getRobotPose(pose2d)) {
```
[the remark this answers is not in the document]
Comment (crdelsey) [reply src/control/nav2_controller_dwb/src/dwb_controller.cpp]: I was originally going to do that, but ended up just making minimal changes to get everything in place in time.

File: src/control/nav2_controller_dwb/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+ament_target_dependencies(${library_name}
+  ${dependencies}
+)
+target_link_libraries(${library_name} ${Boost_LIBRARIES})
+
+ament_target_dependencies(${executable_name}
+  ${dependencies}
+)
+target_link_libraries(${executable_name} ${library_name} ${Boost_LIBRARIES})
```
[the remark this answers is not in the document]
Comment (crdelsey) [reply src/control/nav2_controller_dwb/CMakeLists.txt]: I think @bpwilcox is the one that pointed this out to me. I opened issue #126 for this.

File: src/control/nav2_controller_dwb/src/dwb_controller.cpp [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+        publishZeroVelocity();
+      } else {
+        if (isGoalReached(pose2d)) {
+          break;
+        }
+        auto velocity = odom_sub_->getTwist();
```
[the remark this answers is not in the document]
Comment (crdelsey) [reply src/control/nav2_controller_dwb/src/dwb_controller.cpp]: Agree.

File: src/control/nav2_controller_dwb/CMakeLists.txt [hunk 6]
```diff
@@ -17,9 +17,11 @@ find_package(nav2_tasks REQUIRED)
 find_package(nav2_robot REQUIRED)
 find_package(nav2_planning_msgs REQUIRED)
 find_package(dwb_core REQUIRED)
+find_package(Boost REQUIRED COMPONENTS system thread)
```
Comment (bpwilcox) [inline src/control/nav2_controller_dwb/CMakeLists.txt]: I'm going to remove the Boost dependencies in my next PR for costmap_2d, so we should be able to remove this soon.

**Pull request merged**